### PR TITLE
Speakers filter by name and sorting

### PR DESCRIPTION
**Pull request by beluamat29**

### CODE REVIEW

File: app/models/event_speaker.rb [hunk 1]
```diff
@@ -77,6 +77,16 @@ def profile_image_url
         'https://res.cloudinary.com/dnvnnka50/image/upload/v1552070836/faceless-avatar_yy006o.jpg'
   end
 
+  def self.search(query_string, conference)
+    search_param = "%#{query_string}%"
+
+    speakers = EventSpeaker.
```
Comment (matiasmelendi) [inline app/models/event_speaker.rb]: can we remove this local variable?

Comment (beluamat29) [reply app/models/event_speaker.rb]: sure

File: app/controllers/speakers_controller.rb [hunk 2]
```diff
@@ -1,10 +1,32 @@
 class SpeakersController < ApplicationController
 
   def index
-    @speakers = EventSpeaker.by_conference(@current_conference.campaign)
+    if params[:q].present?
+      @speakers = EventSpeaker.search(params[:q], @current_conference.campaign)
+      @paginate = false
```
Comment (matiasmelendi) [inline app/controllers/speakers_controller.rb]: let's `@paginate = false` for all the possible cases

Comment (beluamat29) [reply app/controllers/speakers_controller.rb]: okey !

File: app/controllers/speakers_controller.rb [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
+    else
+      @speakers = EventSpeaker.by_conference(@current_conference.campaign)
+    end
+
+    if params[:sort_by].present? && (params[:sort_by].eql? 'company')
+      @speakers = @speakers.sort_by { |s| s.heroku_contact_s_account__c }
```
Comment (matiasmelendi) [inline app/controllers/speakers_controller.rb]: @speakers.sort_by(&:heroku_contact_s_account__c)

File: app/controllers/speakers_controller.rb [hunk 4]
```diff
@@ -1,10 +1,32 @@
...
+    end
+    @speakers
   end
 
   def show
     @speaker = EventSpeaker.find(params[:id])
   end
+
+  def search
```
Comment (matiasmelendi) [inline app/controllers/speakers_controller.rb]: can we delete that?

File: app/controllers/speakers_controller.rb (hunk 3 above)
Comment (beluamat29) [reply app/controllers/speakers_controller.rb]: :+1:

File: app/controllers/speakers_controller.rb (hunk 4 above)
Comment (beluamat29) [reply app/controllers/speakers_controller.rb]: done

File: app/controllers/speakers_controller.rb [hunk 5]
```diff
@@ -1,10 +1,32 @@
...
+    @speakers = EventSpeaker.search(params[:q], @current_conference.campaign)
+    @paginate = false
+    render :index
+  end
+
+  private
+
+  def speakers_params
```
Comment (matiasmelendi) [inline app/controllers/speakers_controller.rb]: should we include `q` and use it on `index`?

File: app/models/event_speaker.rb [hunk 6]
```diff
@@ -34,6 +34,7 @@ class EventSpeaker < ApplicationRecord
   scope :most_recent, -> { order(id: :desc) }
   scope :alphabetical, -> { order(heroku_contact_name__c: :asc) }
   scope :featured, -> { where(featured: true) }
+  scope :sort_by_company, -> { order_by { |s| s.heroku_contact_s_account__c } }
```
Comment (matiasmelendi) [inline app/models/event_speaker.rb]: should we delete that?

File: config/routes.rb [hunk 7]
```diff
@@ -41,6 +41,7 @@
     get '/agenda', to: 'agenda#index', as: 'agenda'
     get 'agenda/search', to: 'agenda#search'
     get 'exhibitors/search', to: 'exhibitors#search'
+    get 'speakers/search', to: 'speakers#search'
```
Comment (matiasmelendi) [inline config/routes.rb]: can we remove that?

File: app/models/event_speaker.rb (hunk 6 above)
Comment (beluamat29) [reply app/models/event_speaker.rb]: sure

File: config/routes.rb (hunk 7 above)
Comment (beluamat29) [reply config/routes.rb]: deleted :+1:

File: app/controllers/speakers_controller.rb (hunk 5 above)
Comment (beluamat29) [reply app/controllers/speakers_controller.rb]: :ok_hand:

**Pull request merged**